### ISSUE TITLE
Infinite hang in space_after_comma for parenthesized lists

### ISSUE THREAD

**Issue by asottile** [causes/scss-lint#345]
Simplified test.scss:

```
$z-list: (
    (
        name1
    ),
    (
        name2,
    )
);
```

```
$ git show | head -n 1
commit 30eac23140e26db75d2cde7762ad142286cb6ca8
```

Here's what I get when I ^C

```
$ scss-lint test.scss 
^C/home/asottile/workspace/scss-lint/rbenv/gems/gems/scss-lint-0.32.0/lib/scss_lint/linter/space_after_comma.rb:97:in `block in find_comma_offset': Interrupt
	from /home/asottile/workspace/scss-lint/rbenv/gems/gems/scss-lint-0.32.0/lib/scss_lint/linter/space_after_comma.rb:95:in `loop'
	from /home/asottile/workspace/scss-lint/rbenv/gems/gems/scss-lint-0.32.0/lib/scss_lint/linter/space_after_comma.rb:95:in `find_comma_offset'
	from /home/asottile/workspace/scss-lint/rbenv/gems/gems/scss-lint-0.32.0/lib/scss_lint/linter/space_after_comma.rb:69:in `block in check_commas_after_args'
	from /home/asottile/workspace/scss-lint/rbenv/gems/gems/scss-lint-0.32.0/lib/scss_lint/linter/space_after_comma.rb:68:in `each'
	from /home/asottile/workspace/scss-lint/rbenv/gems/gems/scss-lint-0.32.0/lib/scss_lint/linter/space_after_comma.rb:68:in `check_commas_after_args'
	from /home/asottile/workspace/scss-lint/rbenv/gems/gems/scss-lint-0.32.0/lib/scss_lint/linter/space_after_comma.rb:27:in `visit_script_listliteral'
	from /home/asottile/workspace/scss-lint/rbenv/gems/gems/sass-3.4.10/lib/sass/tree/visitors/base.rb:36:in `visit'
	from /home/asottile/workspace/scss-lint/rbenv/gems/gems/scss-lint-0.32.0/lib/scss_lint/linter.rb:127:in `visit'
	from /home/asottile/workspace/scss-lint/rbenv/gems/gems/scss-lint-0.32.0/lib/scss_lint/linter.rb:138:in `block in visit_children'
	from /home/asottile/workspace/scss-lint/rbenv/gems/gems/scss-lint-0.32.0/lib/scss_lint/linter.rb:136:in `each'
	from /home/asottile/workspace/scss-lint/rbenv/gems/gems/scss-lint-0.32.0/lib/scss_lint/linter.rb:136:in `visit_children'
	from /home/asottile/workspace/scss-lint/rbenv/gems/gems/sass-3.4.10/lib/sass/tree/visitors/base.rb:38:in `visit'
	from /home/asottile/workspace/scss-lint/rbenv/gems/gems/scss-lint-0.32.0/lib/scss_lint/linter.rb:127:in `visit'
	from /home/asottile/workspace/scss-lint/rbenv/gems/gems/scss-lint-0.32.0/lib/scss_lint/linter.rb:138:in `block in visit_children'
	from /home/asottile/workspace/scss-lint/rbenv/gems/gems/scss-lint-0.32.0/lib/scss_lint/linter.rb:136:in `each'
	from /home/asottile/workspace/scss-lint/rbenv/gems/gems/scss-lint-0.32.0/lib/scss_lint/linter.rb:136:in `visit_children'
	from /home/asottile/workspace/scss-lint/rbenv/gems/gems/sass-3.4.10/lib/sass/tree/visitors/base.rb:38:in `visit'
	from /home/asottile/workspace/scss-lint/rbenv/gems/gems/scss-lint-0.32.0/lib/scss_lint/linter.rb:127:in `visit'
	from /home/asottile/workspace/scss-lint/rbenv/gems/gems/scss-lint-0.32.0/lib/scss_lint/linter.rb:21:in `run'
	from /home/asottile/workspace/scss-lint/rbenv/gems/gems/scss-lint-0.32.0/lib/scss_lint/runner.rb:58:in `run_linter'
	from /home/asottile/workspace/scss-lint/rbenv/gems/gems/scss-lint-0.32.0/lib/scss_lint/runner.rb:39:in `block in find_lints'
	from /home/asottile/workspace/scss-lint/rbenv/gems/gems/scss-lint-0.32.0/lib/scss_lint/runner.rb:37:in `each'
	from /home/asottile/workspace/scss-lint/rbenv/gems/gems/scss-lint-0.32.0/lib/scss_lint/runner.rb:37:in `find_lints'
	from /home/asottile/workspace/scss-lint/rbenv/gems/gems/scss-lint-0.32.0/lib/scss_lint/runner.rb:21:in `block in run'
	from /home/asottile/workspace/scss-lint/rbenv/gems/gems/scss-lint-0.32.0/lib/scss_lint/runner.rb:20:in `each'
	from /home/asottile/workspace/scss-lint/rbenv/gems/gems/scss-lint-0.32.0/lib/scss_lint/runner.rb:20:in `run'
	from /home/asottile/workspace/scss-lint/rbenv/gems/gems/scss-lint-0.32.0/lib/scss_lint/cli.rb:52:in `scan_for_lints'
	from /home/asottile/workspace/scss-lint/rbenv/gems/gems/scss-lint-0.32.0/lib/scss_lint/cli.rb:46:in `act_on_options'
	from /home/asottile/workspace/scss-lint/rbenv/gems/gems/scss-lint-0.32.0/lib/scss_lint/cli.rb:25:in `run'
	from /home/asottile/workspace/scss-lint/rbenv/gems/gems/scss-lint-0.32.0/bin/scss-lint:6:in `<top (required)>'
	from ./rbenv/gems/bin/scss-lint:23:in `load'
	from ./rbenv/gems/bin/scss-lint:23:in `<main>'
```

**Comment by asottile** [comment]
I think I've figured this one out, I'll make a PR for it

**Comment by asottile** [comment]
Actually never mind, I'm kinda lost as to what is causing this.  What I have noticed is the variable is actually a list of (mixed) strings and lists.

**Comment by sds** [comment]
Fixed by #346. Thanks for the report!
